### ISSUE TITLE
Cache directory when using XDG_CACHE_ROOT is incorrect

### ISSUE THREAD

**Issue by davefp** [rubocop-hq/rubocop#6306]
## Expected behavior

When setting the `XDG_CACHE_HOME` env var I expect rubocop to cache in `XDG_CACHE_HOME/rubocop_cache` as specified in the docs: https://github.com/rubocop-hq/rubocop/blob/master/manual/caching.md#cache-path

## Actual behavior

Instead it caches in `XDG_CACHE_HOME/<some_generated_number>/rubocop_cache`

## Steps to reproduce the problem

```
> XDG_CACHE_HOME=.mycache rubocop
...
> ls -al .mycache
drwxr-xr-x   3 david  staff    96 18 Sep 15:52 .
drwxr-xr-x  49 david  staff  1568 18 Sep 15:52 ..
drwxr-xr-x   3 david  staff    96 18 Sep 15:52 501
```

## RuboCop version

```
0.52.1 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-darwin17)
```

**Comment by koic** [comment]
Thanks for the feedback.

> When setting the XDG_CACHE_HOME env var I expect rubocop to cache in `XDG_CACHE_HOME/rubocop_cache`

Actualy, it will be the path that contains the user ID of the process.
https://github.com/rubocop-hq/rubocop/blob/v0.59.1/lib/rubocop/result_cache.rb#L69

`$XDG_CACHE_HOME/<Process.uid>/rubocop_cache`

I think this is a document mistake.

**Comment by koic** [comment]
The mistake in this document has been resolved with #6350.
